### PR TITLE
feat: save user agreements on create and update registration mail

### PR DESCRIPTION
**Pull request by ReidyT**
This PR is linked to graasp/graasp-auth#357.

This includes:
- Add the user agreements links as a reminder in the registration mail.
- Add the `userAgreements` field with column name `user_agreements`.

Here are the updated mails in english and french (other langs will display in english):

### CODE REVIEW

File: src/services/member/repository.ts [hunk 1]
```diff
@@ -83,7 +83,7 @@ export const MemberRepository = AppDataSource.getRepository(Member).extend({
 
   async post(data: Partial<Member> & Pick<Member, 'email'>): Promise<Member> {
     const email = data.email.toLowerCase();
-    const createdMember = await this.insert({ ...data, email });
+    const createdMember = await this.insert({ ...data, email, userAgreements: new Date() });
```
Comment (pyphilia) [inline src/services/member/repository.ts:86]: I would add a comment that we suppose any new member accepted the agreements.

File: src/plugins/mailer/langs/en.json [hunk 2]
```diff
@@ -24,5 +24,6 @@
   "SHARE_ITEM_BUTTON": "View Item",
   "SHARE_ITEM_TITLE": "{{creatorName}} has shared {{itemName}} with you on Graasp",
   "FOOTER": "You are receiving this email because of your account on <a href=\"https://graasp.org\">Graasp</a>",
-  "POWERED_BY": "Powered by <a href=\"https://graasp.org\">Graasp</a>."
+  "POWERED_BY": "Powered by <a href=\"https://graasp.org\">Graasp</a>.",
+  "USER_AGREEMENTS_MAIL_TEXT": "* By clicking on <strong>{{SIGN_UP_BUTTON_TEXT}}</strong>, you agree to the <a href=\"{{GRAASP_LANDING_PAGE_ORIGIN}}/terms\">terms of service</a> and the <a href=\"{{GRAASP_LANDING_PAGE_ORIGIN}}/privacy\">privacy policy</a>."
```
Comment (pyphilia) [inline src/plugins/mailer/langs/en.json:28]:
```suggestion
  "USER_AGREEMENTS_MAIL_TEXT": "* By clicking on <strong>{{signUpButtonText}}</strong>, you agree to the <a href=\"{{GraaspLandingPageOrigin}}/terms\">terms of service</a> and the <a href=\"{{GraaspLandingPageOrigin}}/privacy\">privacy policy</a>."
```

File: src/plugins/mailer/langs/fr.json [hunk 3]
```diff
@@ -24,5 +24,6 @@
   "SHARE_ITEM_BUTTON": "Voir l'élément",
   "SHARE_ITEM_TITLE": "{{creatorName}} a partagé l'élément {{itemName}} avec vous sur Graasp",
   "FOOTER": "Cet email vous a été envoyé car vous avez un compte sur <a href=\"https://graasp.org\">Graasp</a>",
-  "POWERED_BY": "Conçu par <a href=\"https://graasp.org\">Graasp</a>."
+  "POWERED_BY": "Conçu par <a href=\"https://graasp.org\">Graasp</a>.",
+  "USER_AGREEMENTS_MAIL_TEXT": "* En cliquant sur <strong>{{SIGN_UP_BUTTON_TEXT}}</strong>, vous acceptez les <a href=\"{{GRAASP_LANDING_PAGE_ORIGIN}}/fr/terms\">conditions d'utilisation</a> et la <a href=\"{{GRAASP_LANDING_PAGE_ORIGIN}}/fr/privacy\">politique de confidentialité</a>."
```
Comment (pyphilia) [inline src/plugins/mailer/langs/fr.json:28]: same as above.

File: src/services/auth/service.ts [hunk 4]
```diff
@@ -59,10 +60,19 @@ export class AuthService {
...
-    ${this.mailer.buildText(translated(MAIL.SIGN_UP_TEXT))}
+    ${this.mailer.buildText(greetingsAndSignupText)}
     ${this.mailer.buildButton(link, translated(MAIL.SIGN_UP_BUTTON_TEXT))}
+    ${this.mailer.buildText(
+      translated(MAIL.USER_AGREEMENTS_MAIL_TEXT, {
+        SIGN_UP_BUTTON_TEXT: translated(MAIL.SIGN_UP_BUTTON_TEXT),
+        GRAASP_LANDING_PAGE_ORIGIN: GRAASP_LANDING_PAGE_ORIGIN,
+      }),
+      // Add margin top of -15px to remove 15px margin bottom of the button.
+      // TODO: If the margin bottom of the button can be removed in "layout.ts", remove the style here.
```
Comment (pyphilia) [inline src/services/auth/service.ts:73]: Did you check the todo? But I guess it's okay to keep -15px, it's not too dramatic.

**Review by pyphilia — CHANGES_REQUESTED** [review]
Thanks for the PR, really clean 🦜

File: src/services/member/repository.ts (hunk 1 above)
Comment (ReidyT) [reply src/services/member/repository.ts:86]: Yes it's a really good idea !

File: src/services/auth/service.ts (hunk 4 above)
Comment (ReidyT) [reply src/services/auth/service.ts:73]: No, I forgot to discuss with you about that. I also think it is okay, so I will remove the TODO ;).

File: src/services/member/repository.ts [hunk 5]
```diff
@@ -83,7 +83,12 @@ export const MemberRepository = AppDataSource.getRepository(Member).extend({
 
   async post(data: Partial<Member> & Pick<Member, 'email'>): Promise<Member> {
     const email = data.email.toLowerCase();
-    const createdMember = await this.insert({ ...data, email });
+
+    // The backend assumes user agrees to terms by creating an account.
+    // The auth frontend only block the user to create an account without checking the boxes.
```
Comment (pyphilia) [inline src/services/member/repository.ts:88]:
```suggestion
    // The auth frontend only blocks the user to create an account without checking the boxes.
```

**Review by pyphilia — APPROVED** [review]

**Pull request merged**